### PR TITLE
refactor: use single event emitter for all events

### PR DESCRIPTION
**Pull request by TimoGlastra**
Signed-off-by: Timo Glastra <timo@animo.id>

### CODE REVIEW

**Review by jakubkoci — COMMENTED** [review]
I think it's a good step forward in a way how we want to use events in the framework 👍  Maybe, I would reconsider replacing strings as event names. It seems it won't be possible to use the interface for that, right?

File: src/modules/connections/ConnectionEvents.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { ConnectionRecord, ConnectionState } from '.'
```
[the remark this answers is not in the document]
Comment (TimoGlastra) [reply src/modules/connections/ConnectionEvents.ts:1]: It probably does....

I fiddled around with `import/no-cycle` eslint rule (https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-cycle.md). We have a lot of cyclic dependencies so we're going to need to do quite some refactoring to remove them all. If it is even possible. Sometimes it's unavoidable to have cyclic dependencies. 

I think a problem with `import/no-cycle` that it doesn't take TS type only imports into account. We sometimes just have cyclic dependencies due to type imports.

File: src/modules/connections/ConnectionEvents.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+
+export interface ConnectionStateChangedEvent {
```
[the remark this answers is not in the document]
Comment (TimoGlastra) [reply src/modules/connections/ConnectionEvents.ts:3]: Nope just not paying attention :)

**Comment by TimoGlastra** [comment]
Updated according to your feedback @jakubkoci

**Review by jakubkoci — APPROVED** [review]

**Comment by JamesKEbert** [comment]
Question--were we able to test this in React Native? I know I've ran into issues with the Event Emitter before in RN.

**Pull request merged**